src/providers/idp/idp_id.c

```diff
@@ -0,0 +1,698 @@
+/*
+    SSSD
+
+    IdP Identity Backend Module
+
+    Authors:
+        Sumit Bose <sbose@redhat.com>
+
+    Copyright (C) 2024 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <errno.h>
+
+#include "util/util.h"
+#include "util/sss_chain_id.h"
+#include "providers/idp/idp_id.h"
+
+enum idp_lookup_type {
+    IDP_LOOKUP_USER,
+    IDP_LOOKUP_GROUP,
+    IDP_LOOKUP_USER_GROUPS,
+    IDP_LOOKUP_GROUP_MEMBERS
+};
+
+static
+errno_t set_oidc_extra_args(TALLOC_CTX *mem_ctx, struct idp_id_ctx *idp_id_ctx,
```

Comment:
This is so similar to `set_oidc_auth_extra_args()` that it begs for unification.